gateway/single-cluster/regional-l7-ilb/README.md

```diff
@@ -0,0 +1,259 @@
+# Single-cluster Gateway with Regional L7 Internal Load Balancing
+
+This recipe provides a basic walk through for setting up Single-cluster Gateway with the `gke-l7-rilb` GatewayClass, provisioning an regional internal HTTP/S Load Balancer.
+
+To achieve this, we will:
+
+- Deploy v1 of a sample application into the GKE cluster named `gke-1`, as well as its ServiceExport.
+- Deploy a Gateway using the `gke-l7-rilb` single-cluster GatewayClass to the GKE cluster named `gke-1`, serving as this example's config cluster.
+- Deploy an HTTPRoute to route external traffic between v1 of the sample application and v2 of the sample application to achieve traffic splitting.
+
+### Relevant documentation
+
+- [Gateway API](https://cloud.google.com/kubernetes-engine/docs/concepts/gateway-api)
+- [Gateway API resources](https://cloud.google.com/kubernetes-engine/docs/concepts/gateway-api#gateway_resources)
+- [Deploying Gateways](https://cloud.google.com/kubernetes-engine/docs/how-to/deploying-gateways)
+- [Proxy-only subnets for internal HTTP(S) load balancers](https://cloud.google.com/load-balancing/docs/l7-internal/proxy-only-subnets)
+
+## Setup
+
+Set the project environment variable and gcloud configuration
+```
+$ export PROJECT_ID=your_project_id
+$ gcloud config set project $PROJECT_ID
+```
+
+Enable the required GCP APIs.
+```
+$ gcloud services enable \
+     container.googleapis.com 
+```
+
+Beware to [create a proxy-only subnet](https://cloud.google.com/load-balancing/docs/l7-internal/proxy-only-subnets#proxy_only_subnet_create) in the same region of the cluster you're going to create
+```
+gcloud compute networks subnets create SUBNET_NAME \
+    --purpose=INTERNAL_HTTPS_LOAD_BALANCER \
+    --role=ACTIVE \
+    --region=REGION \
+    --network=VPC_NETWORK_NAME \
+    --range=CIDR_RANGE
+```
+
+Create one GKE cluster.
```

Comment:
Instead of including the instructions to create a cluster here please refer to the cluster creation instructions in the cluster-setup.md page https://github.com/GoogleCloudPlatform/gke-networking-recipes/blob/master/cluster-setup.md#single-cluster-environment